packages/driver/src/cy/commands/querying.js

```diff
@@ -277,7 +277,17 @@ module.exports = (Commands, Cypress, cy) => {
         let $el
 
         try {
-          $el = cy.$$(selector, options.withinSubject)
+          // only support shadow traversal if we're not searching
+          // within a subject and have been explicitly told to ignore
+          // boundaries.
+          if (!options.ignoreShadowBoundaries || options.withinSubject) {
```

Comment:
Why not traverse into a shadow root if within a subject? Seems like this should be valid (using html from PR description):

```javascript
cy.get('.top-level').within(() => {
  cy.get('.my-button', { ignoreShadowBoundaries: true })
  .click()
})
```